src/libs/highlevel/elektra.c

```diff
@@ -0,0 +1,245 @@
+/**
+ * @file
+ *
+ * @brief Elektra High Level API.
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "elektra.h"
+#include <stdlib.h>
+#include <memory.h>
+#include "kdb.h"
+#include "kdbhelper.h"
+#include "elektraprivate.h"
+
+#define READ_KEY \
+Key * const nameKey = keyDup (elektra->parentKey); \
```

Comment:
Why duplicate? What about having a searchKey in `Elektra` and only modify its name?